File: server/src/domain/catalog/Seller.js
const Uuid = require('../shared/Uuid');
const SellerStatus = require('./SellerStatus');

class Seller {
    constructor({ id, userId, storeName, address, description, status = SellerStatus.PENDING, rejectionReason = null }) {
        this.id = id || Uuid.random();
        this.userId = userId;
        this.storeName = storeName;
        this.address = address;
        this.description = description;
        this.status = status;
        this.rejectionReason = rejectionReason;

        this._validate();
    }

    static create(userId, storeName, address, description) {
        return new Seller({
            userId,
            storeName,
            address,
            description
        });
    }

    approve() {
        this.status = SellerStatus.APPROVED;
        this.rejectionReason = null;
    }

    reject(reason) {
        if (!reason) throw new Error('Rejection reason is required');
        this.status = SellerStatus.REJECTED;
        this.rejectionReason = reason;
    }

    isApproved() {
        return this.status === SellerStatus.APPROVED;
    }

    _validate() {
        if (!this.userId) throw new Error('Seller must be linked to a user');
        if (!this.storeName) throw new Error('Store name is required');
        if (!this.address) throw new Error('Address is required');
    }
}

module.exports = Seller;

File: backend-dotnet/Antigaspi.Domain/Entities/Seller.cs
using Antigaspi.Domain.Enums;
using Antigaspi.Domain.ValueObjects;

namespace Antigaspi.Domain.Entities;

public class Seller
{
    public Guid Id { get; private set; }
    public Guid UserId { get; private set; }
    public string StoreName { get; private set; }
    public Address Address { get; private set; }
    public string Description { get; private set; }
    public SellerStatus Status { get; private set; }
    public string? RejectionReason { get; private set; }

    private Seller() { }

    public Seller(Guid userId, string storeName, Address address, string description, Guid? id = null, SellerStatus status = SellerStatus.PENDING, string? rejectionReason = null)
    {
        if (userId == Guid.Empty) throw new ArgumentException("Seller must be linked to a user");
        if (string.IsNullOrWhiteSpace(storeName)) throw new ArgumentException("Store name is required");
        if (address is null) throw new ArgumentNullException(nameof(address), "Address is required");

        Id = id ?? Guid.NewGuid();
        UserId = userId;
        StoreName = storeName;
        Address = address;
        Description = description;
        Status = status;
        RejectionReason = rejectionReason;
    }

    public static Seller Create(Guid userId, string storeName, Address address, string description)
    {
        return new Seller(userId, storeName, address, description);
    }

    public void UpdateDetails(string storeName, string description, Address address)
    {
        if (string.IsNullOrWhiteSpace(storeName)) throw new ArgumentException("Store name is required");
        if (address is null) throw new ArgumentNullException(nameof(address), "Address is required");

        StoreName = storeName;
        Description = description;
        Address = address;
    }

    public void Approve()
    {
        Status = SellerStatus.APPROVED;
        RejectionReason = null;
    }

    public void Reject(string reason)
    {
        if (string.IsNullOrWhiteSpace(reason)) throw new ArgumentException("Rejection reason is required");
        Status = SellerStatus.REJECTED;
        RejectionReason = reason;
    }

    public bool IsApproved()
    {
        return Status == SellerStatus.APPROVED;
    }
}
